### DISCUSSION TITLE
Finding past weather data for specific timestamp + lat/longitude

### DISCUSSION THREAD

**Discussion by kfeinberg** [weather-gov/api#215] [Q&A]
Hello! I am new to the API + trying to use it to find the past weather data for a specific timestamp and longitude/latitude. Right now I am first finding a weather station through using the https://api.weather.gov/points/${lat},${long} endpoint, then pulling one of the observation stations randomly from the observationStations property (ex: https://api.weather.gov/gridpoints/BOX/63,51/stations). Then I look at the observations for this weather station (ex: https://api.weather.gov/stations/KIND/observations) and use the one with the closest timestamp to my timestamp. Does anyone know of a better way to do this? It doesn't seem super efficient, but I couldn't figure out any better way of doing it. Thank you!!!!

**Comment by hluker** [comment]
Hi @kfeinberg,

I don't think the NWS API is able to access historical data... but don't quote me on that. Maybe try this? http://mesonet.agron.iastate.edu/api/1/docs#/default/service_obhistory__fmt__get. You can have data returned by station and timestamp. Here's an example of the data returned back http://mesonet.agron.iastate.edu/api/1/obhistory.json?network=IA_ASOS&station=MCW&date=2019-03-29&full=true

Henry

**Comment by kfeinberg** [comment]
Thank you! I don't need historical data further back than a week, so I was trying to use the https://api.weather.gov/stations/{observation_station}/observations endpoint (ex: https://api.weather.gov/stations/C0728/observations)

**Comment by hluker** [comment]
Ahhh, gotcha. You could probably narrow your first two requests down to one:
https://api.weather.gov/points/42.5349,-92.4453/stations to automatically get the stations list (redirects)

The closest station is the first entry in the features array. 
so if you got $features[0]->id: i.e. https://api.weather.gov/stations/KALO
You could just add /observations/latest to the end of that to get the latest observation.

As far as I know, you'll need at least two requests: one to get the metadata for a particular lat/lon point, and another to actually retrieve the data for a station
